src/Components/Endpoints/src/Rendering/ViewBufferTextWriter/ViewBufferTextWriter.cs

```diff
@@ -0,0 +1,292 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/*
+ * The implementation here matches the one present at https://github.com/dotnet/aspnetcore/blob/88180f6f487a1222b3af8c111aa6b5f8aa278633/src/Mvc/Mvc.ViewFeatures/src/Buffers/ViewBufferTextWriter.cs
+ * with the distinction that it implements a constructor that takes a concrete `MemoryPoolViewBuffer` type instead of
+ * the `ViewBuffer` abstract type. Source is copied here instead of shared to avoid circular dependency issues
+ * between Mvc.Razor and Components.Endpoints.
+ */
```

Comment:
Do you think we could file an issue tracking eventually merging these two implementations into one?

Reply:
See https://github.com/dotnet/aspnetcore/issues/49932